ssl/statem/statem.c

```diff
@@ -404,7 +442,8 @@ static int state_machine(SSL *s, int server)
             }
         } else {
             /* Error */
-            ossl_statem_set_error(s);
+            check_fatal(s, SSL_F_STATE_MACHINE);
+            SSLerr(SSL_F_STATE_MACHINE, ERR_R_INTERNAL_ERROR);
```

Comment:
Why the internal error?

Reply:
Hmmm...that's probably not appropriate actually. In this case we *could* end up here via internal error - but far more likely is that we've been called incorrectly by the application, i.e. we've been called again after a previous fatal error. I've changed the error code.